packages/vm/src/instance.rs

```diff
@@ -213,11 +216,15 @@ where
             }
         }
 
+        let lock = m.map(|m| m.lock().unwrap());
         let wasmer_instance = Box::from(WasmerInstance::new(module, &import_obj).map_err(
             |original| {
                 VmError::instantiation_err(format!("Error instantiating module: {:?}", original))
             },
         )?);
+        if let Some(lock) = lock {
+            drop(lock)
+        }
```

Comment:
It should be possible to lock this via RAII:

```rust
        let wasmer_instance = Box::from(
            {
                let lock = m.map(|m| m.lock().unwrap());
                WasmerInstance::new(module, &import_obj)
            }
            .map_err(|original| {
                VmError::instantiation_err(format!("Error instantiating module: {:?}", original))
            })?,
        );
  ```

Reply:
Nice. Will do.

Reply:
Come to think of it, this is crying for a `Mutex<WasmerInstance>`. This is the famed "lock data, not code approach"... not sure what the implications will be, i.e. how and when to break out of the lock, etc.